Add externs for iron-list

rictic opened the pull request:
Used to get iron-list compiling.

rictic committed:
Add @jlklein24's extern discoveries!

Used to get iron-list compiling.

jklein24 commented on polymer.externs.js:
```diff
@@ -666,10 +739,84 @@ Polymer.Templatizer = {
+   *   }
+   *
+   * @param {!HTMLElement} el Element for which to return a template model.
+   * @return {Object<PolymerElement>} Model representing the binding scope for
```
Not sure this type is right here (I just copied it from the docs). Do you know what type this actually is? We should also add either ! or ? to the front of both Object and PolymerElement.

rictic replied:
Looks like the intent was to say that the returned object implements or extends Polymer.Base. In our type system I think that most closely corresponds to PolymerElement.

Here's the source to double-check this: https://github.com/Polymer/polymer/blob/8868fdc9f231c615e554ec21833efe0d62de4718/src/lib/template/templatizer.html#L411

jklein24 replied:
Interesting... I think you're right. It does seem weird that "model" is an instance of PolymerElement, though. Shrug. Go with {?PolymerElement}

jklein24 commented:
Feels weird reviewing my own code, but wow, nice code!! LGTM :-p

rictic committed:
Update modelForElement's return type.

Looks like the intent was to say that the returned object implements or extends Polymer.Base. In our type system I think that most closely corresponds to PolymerElement.

Here's the source to double-check this: https://github.com/Polymer/polymer/blob/8868fdc9f231c615e554ec21833efe0d62de4718/src/lib/template/templatizer.html#L411

rictic committed:
modelForElement may return undefined

rictic commented:
As discussed, refined type for modelForElement's return. Merging!
